src/tracking/session-storage.ts

```diff
@@ -0,0 +1,40 @@
+function safeGetSessionStorage(): Storage | undefined {
+  try {
+    return window?.sessionStorage;
+  } catch (err) {
+    // Catch a SecurityError that is thrown if the browser has disabled access
+    // to session storage
+    return undefined;
+  }
+}
+
+// rome-ignore lint/suspicious/noExplicitAny: any is the right type here
+export function getItem(key: string): any {
+  const storage = safeGetSessionStorage();
+  if (!storage) {
+    return undefined;
+  }
+
+  const str = storage.getItem(key);
+  return str ? JSON.parse(str) : str;
+}
+
+// rome-ignore lint/suspicious/noExplicitAny: any is the right type here
```

Comment:
You can remove the duplication by doing something like `type AnyJSON = any;` and then reusing that.